### PR TITLE
Support for file operations

### PR DESCRIPTION
**Pull request by seba-aln**
feat: Added support for file sharing operations

### CODE REVIEW

File: examples/FileSharing.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+//     ->fileName($fileName)
+//     ->message("Hello from PHP SDK")
+//     ->fileHandle($fileHandle)
+//     ->sync();
+
+// fclose($fileHandle);
+
+// Listing files in the channel
+$channelFiles = $pubnub->listFiles()->channel($channelName)->sync();
+if ($channelFiles->getCount() > 0) {
```
Comment (marcin-cebo) [inline examples/FileSharing.php:31]: What do you think about introducing:
$fileCount = $channelFiles->getCount();
?

File: src/PubNub/Endpoints/FileSharing/PublishFileMessage.php [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            $this->pubnub->getConfiguration()->getSubscribeKey(),
+            urlencode($this->channel),
+            urlencode($this->buildMessage())
+        );
+    }
+
+    public function encryptMessage($message)
+    {
+        $crypto = $this->pubnub->getCryptoSafe();
+        $message = PubNubUtil::writeValueAsString($message);
```
Comment (marcin-cebo) [inline src/PubNub/Endpoints/FileSharing/PublishFileMessage.php:59]: What do you think about renaming to:
 $messageString
?

File: src/PubNub/Endpoints/FileSharing/PublishFileMessage.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        if ($crypto) {
+            return $crypto->encrypt($message);
+        }
+        return $message;
+    }
+
+    protected function buildMessage()
+    {
+        $message = [
```
Comment (marcin-cebo) [inline src/PubNub/Endpoints/FileSharing/PublishFileMessage.php:68]: Wouldn't  `$messageData` be a more descriptive name?

File: src/PubNub/Endpoints/FileSharing/SendFile.php [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+
+
+    /**
+     * @return array
+     */
+    protected function customParams()
+    {
+        $params = [];
+        foreach ($this->customParamMapping as $customParam => $requestParam) {
+            if (isset($this->$customParam) && !empty($this->$customParam)) {
```
Comment (marcin-cebo) [inline src/PubNub/Endpoints/FileSharing/SendFile.php:172]: Is `issset` required here? What is a responsibility of `empty` method check?

File: src/PubNub/Endpoints/FileSharing/SendFile.php [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+        } catch (\Exception $e) {
+            throw $e;
+        }
+
+        $publishRequest = new PublishFileMessage($this->pubnub);
+        $publishRequest->channel($this->channel)
+            ->fileId($this->fileUploadEnvelope->getFileId())
+            ->fileName($this->fileName);
+
+        if (isset($this->meta)) {
```
Comment (marcin-cebo) [inline src/PubNub/Endpoints/FileSharing/SendFile.php:266]: This line is duplicated in line 269, 272, 275 is this intentional?

File: examples/FileSharing.php (hunk 1 above)
Comment (seba-aln) [reply examples/FileSharing.php:31]: would improve readability

File: src/PubNub/Endpoints/FileSharing/PublishFileMessage.php (hunk 3 above)
Comment (seba-aln) [reply src/PubNub/Endpoints/FileSharing/PublishFileMessage.php:68]: fixed

File: src/PubNub/Endpoints/FileSharing/SendFile.php (hunk 4 above)
Comment (seba-aln) [reply src/PubNub/Endpoints/FileSharing/SendFile.php:172]: `isset` is a safeguard for accessing unintialized typed property and `!empty` checks if the value is not empty (not null, `0` or empty string

File: src/PubNub/Endpoints/FileSharing/PublishFileMessage.php (hunk 2 above)
Comment (seba-aln) [reply src/PubNub/Endpoints/FileSharing/PublishFileMessage.php]: renamed

**Review by marcin-cebo — APPROVED** [review]

**Comment by seba-aln** [comment]
@pubnub-release-bot release

**Review by techwritermat — APPROVED** [review]

**Pull request merged**